network/transport/types.go

```diff
@@ -41,6 +43,14 @@ func (p PeerID) String() string {
 	return string(p)
 }
 
+// PeerKey is a unique key for a Peer including its PeerID, NodeDID, and address.
+type PeerKey string
+
+// String returns the PeerKey as string.
+func (p PeerKey) String() string {
```

Comment:
not a very useful type?

Reply:
The changes to diagnostics manager resulted in some return values changing from `map[PeerID]any` to `map[string]any`. I didn't likt the ambiguity there so added the `PeerKey`